plugins/SegmentEditor/API.php

```diff
@@ -407,4 +433,16 @@ private function getMessageCannotEditSegmentCreatedBySuperUser()
 
         return $message;
     }
+
+    private function reArchiveSegment($segmentInfo)
+    {
+        $definition = $segmentInfo['definition'];
+        $idSite = $segmentInfo['enable_only_idsite'] ?? 'all';
+
+        $idSites = Site::getIdSitesFromIdSitesString($idSite);
```

Comment:
shouldn't that be done as super user? Otherwise it might only rearchive those sites the current use has access for.

Reply:
changed